util/vizerror/vizerror.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package vizerror provides types and utility funcs for handling visible errors
+// that are safe to display to end users.
+package vizerror
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Error is an error that is safe to display to end users.
+type Error error
+
+// New returns an error that formats as the given text.
+func New(text string) error {
+	return Error(errors.New(text))
+}
+
+// Errorf returns an error with the specified format and values.
```

Comment:
Maybe on both of these document that the returned type is guaranteed to be of type Error?

Reply:
Any reason not to just return an `Error` to make that really explicit?  I know it's recommended to return concrete types from functions rather than interfaces, but I'm not sure if errors are treated differently.

Reply:
Because that's the convention (e.g. look at all the errors returned by the `os` package) Otherwise people fall into https://go.dev/doc/faq#nil_error ... 

Or do stuff like:

```go
     err := vizerror.New("foo")
     if thingsGood {
          err = nil
      }
```

And are surprised it doesn't compile. Or the type is a pointer, find it compiles but hit that FAQ problem

Reply:
done